snackpub/src/RateLimiter.ts

```diff
@@ -0,0 +1,63 @@
+import type { createClient } from 'redis';
+
+interface RateLimiterOptions {
+  maxOperations: number;
+  intervalSeconds: number;
+}
+
+export default class RateLimiter {
+  constructor(private readonly redisClient: ReturnType<typeof createClient>) {}
+
+  public hasExceededSrcIpRateAsync(
+    srcIp: string,
+    _socketId: string,
+    options: RateLimiterOptions = { maxOperations: 30, intervalSeconds: 60 } // 30 connections per IP per hour
+  ): Promise<boolean> {
+    return this.hasExceededKeyAsync(srcIp, `ip:${srcIp}`, options);
+  }
+
+  public hasExceededMessagesRateAsync(
+    srcIp: string,
+    socketId: string,
+    options: RateLimiterOptions = { maxOperations: 600, intervalSeconds: 1 } // 600 messages per socket.id per second
```

Comment:
600 messages per second per source IP seems like a lot. What is the use case for this many messages?

We could also set up two rate limits: 600/second and 6000/minute. This would allow for high numbers of messages at once, but not continuously.